src/main/java/org/springframework/metrics/instrument/prometheus/PrometheusTimer.java

```diff
@@ -16,31 +16,32 @@
 package org.springframework.metrics.instrument.prometheus;
 
 import org.springframework.metrics.instrument.Clock;
+import org.springframework.metrics.instrument.Measurement;
+import org.springframework.metrics.instrument.Tag;
 import org.springframework.metrics.instrument.internal.AbstractTimer;
+import org.springframework.metrics.instrument.internal.MeterId;
 import org.springframework.metrics.instrument.stats.quantile.Quantiles;
 
+import java.util.Collections;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 import static org.springframework.metrics.instrument.internal.TimeUtils.secondsToUnit;
 
 public class PrometheusTimer extends AbstractTimer {
     private CustomPrometheusSummary.Child summary;
-    private Quantiles quantiles;
 
-    public PrometheusTimer(String name, CustomPrometheusSummary.Child summary, Clock clock, Quantiles quantiles) {
-        super(name, clock);
+    PrometheusTimer(MeterId id, CustomPrometheusSummary.Child summary, Clock clock) {
+        super(id, clock);
         this.summary = summary;
-        this.quantiles = quantiles;
```

Comment:
Rather moved down into the custom summary. This was a holdover from an earlier attempt, and I goofed by not removing it here. Results in the value being observed to the quantiles multiple times per.